zerver/lib/message.py

```diff
@@ -1282,6 +1291,94 @@ def remove_message_id_from_unread_mgs(state: RawUnreadMessagesResult, message_id
     state["mentions"].discard(message_id)
 
 
+def format_unread_message_details(
+    user_id: int,
+    raw_unread_data: RawUnreadMessagesResult,
+) -> Dict[str, Dict[str, Any]]:
+    unread_data = {}
+
+    for message_id, private_message_details in raw_unread_data["pm_dict"].items():
+        other_user_id = private_message_details["other_user_id"]
+        if other_user_id == user_id:
+            user_ids = []
```

Comment:
Let's put a brief comment for this little block that we exclude our own user ID, so the client knows that.